install_scripts/templates/common/kubernetes-upgrade.sh

```diff
@@ -423,6 +437,96 @@ upgradeK8sWorkers() {
     spinnerNodesReady
 }
 
+#######################################
+# Upgrade Kubernetes on remote masters to version. Never downgrades a worker.
+# Globals:
+#   AIRGAP
+# Arguments:
+#   k8sVersion - e.g. 1.10.6
+#   shouldUpgradePatch
+# Returns:
+#   None
+#######################################
+upgradeK8sRemoteMasters() {
+    k8sVersion="$1"
+    shouldUpgradePatch="$2"
+
+    semverParse "$k8sVersion"
+    local upgradeMajor="$major"
+    local upgradeMinor="$minor"
+    local upgradePatch="$patch"
+
+    local nodes=
+    n=0
+    while ! nodes="$(kubectl get nodes --show-labels 2>/dev/null)"; do
+        n="$(( $n + 1 ))"
+        if [ "$n" -ge "30" ]; then
+            # this should exit script on non-zero exit code and print error message
+            nodes="$(kubectl get nodes --show-labels)"
+        fi
+        sleep 2
+    done
+    # not an error if there are no remote masters
+    local masters="$(echo "$nodes" | sed '1d' | grep "node-role.kubernetes.io/master" || :)"
+
+    while read -r node; do
+        if [ -z "$node" ]; then
+            continue
+        fi
+        nodeName=$(echo "$node" | awk '{ print $1 }')
+        if [ "$nodeName" = "$(hostname)" ]; then
+            continue
+        fi
+        nodeVersion="$(echo "$node" | awk '{ print $5 }' | sed 's/v//' )"
+        semverParse "$nodeVersion"
+        nodeMajor="$major"
+        nodeMinor="$minor"
+        nodePatch="$patch"
+        if [ "$nodeMajor" -ne "$upgradeMajor" ]; then
+            printf "Cannot upgrade from %s to %s\n" "$nodeVersion" "$k8sVersion"
+            return 1
+        fi
+        if [ "$nodeMinor" -gt "$upgradeMinor" ]; then
+            continue
+        fi
+        if [ "$nodeMinor" -eq "$upgradeMinor" ] && [ "$nodePatch" -ge "$upgradePatch" ]; then
+            continue
+        fi
+        if [ "$nodeMinor" -eq "$upgradeMinor" ] && [ "$shouldUpgradePatch" != "1" ]; then
+            continue
+        fi
+        kubectl drain "$nodeName" \
+            --delete-local-data \
+            --ignore-daemonsets \
+            --force \
+            --grace-period=30 \
+            --timeout=300s \
+            --pod-selector 'app notin (rook-ceph-mon,rook-ceph-osd,rook-ceph-osd-prepare,rook-ceph-operator,rook-ceph-agent)' || true
```

Comment:
why do you ignore the failure here?

Reply:
Sometimes draining hangs. The drain was added because in testing it made Docker less likely to get hung up. We still want to attempt the upgrade if the drain fails.